File: skills/astra-toolkit/clients/python/examples/tables-alter-type-rename.py
import os
from astrapy import DataAPIClient
from astrapy.info import (
    AlterTypeRenameFields,
)

# Get a database
client = DataAPIClient()
database = client.get_database(
    os.environ["ASTRA_DB_API_ENDPOINT"], token=os.environ["ASTRA_DB_APPLICATION_TOKEN"]
)

# Rename fields in a user-defined type
database.alter_type(
    "member",
    AlterTypeRenameFields(
        fields={"name": "first_name", "is_active": "is_member"}
    ),
)

File: skills/astra-toolkit/clients/java/examples/tables-alter-type-rename.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.commands.AlterTypeRenameFields;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient(System.getenv("ASTRA_DB_APPLICATION_TOKEN")).getDatabase(System.getenv("ASTRA_DB_API_ENDPOINT"));

    // Rename fields in a user-defined type
    database.alterType(
        "member",
        new AlterTypeRenameFields()
            .addField("name", "first_name")
            .addField("is_active", "is_member"));
  }
}
